kedro/framework/cli/project.py

```diff
@@ -372,6 +374,14 @@ def run(
         )
     runner = runner or "SequentialRunner"
     if parallel:
+        deprecation_message = (
+            "DeprecationWarning: The behaviour of --parallel and -p flags will change. "
```

Comment:
Do we want to mention the thinking that we want to `-p` to fall under the `--pipeline` category?

Reply:
Already there in the next line 😀